src/types/table.ts

```diff
@@ -13,6 +15,8 @@ export interface TablePreferences {
   columnVisibility?: Record<string, boolean>
 }
 
+export type TableData = Array<Record<string, any>>
```

Comment:
issue: `TableData` interface name is a bit misleading here since we have multiple table components, but only applies to this new component. This should be renamed `TableViewData` and update the docs/usage accordingly.